packages/sit-onyx/src/components/OnyxListbox/OnyxListbox.vue

```diff
@@ -1,19 +1,43 @@
 <script lang="ts" setup generic="TValue extends SelectionOptionValue = SelectionOptionValue">
 import { createListbox } from "@sit-onyx/headless";
-import { computed, ref, watch } from "vue";
+import plusSmall from "@sit-onyx/icons/plus-small.svg?raw";
+import { computed, ref, watch, watchEffect } from "vue";
+import { useScrollEnd } from "../../composables/scrollEnd";
+import { injectI18n } from "../../i18n";
+import OnyxButton from "../OnyxButton/OnyxButton.vue";
+import OnyxEmpty from "../OnyxEmpty/OnyxEmpty.vue";
 import OnyxListboxOption from "../OnyxListboxOption/OnyxListboxOption.vue";
+import OnyxLoadingIndicator from "../OnyxLoadingIndicator/OnyxLoadingIndicator.vue";
 import type { SelectionOptionValue } from "../OnyxRadioButton/types";
 import type { OnyxListboxProps } from "./types";
 
-const props = defineProps<OnyxListboxProps<TValue>>();
+const props = withDefaults(defineProps<OnyxListboxProps<TValue>>(), {
+  loading: false,
```

Comment:
```suggestion
  loading: false,
  loadingMode: {},
```

this might simplify a lot of the logic below which checks if it is an object because undefined can not happen anymore